core/src/main/java/org/apache/metamodel/DeleteAndInsertBuilder.java

```diff
@@ -103,7 +103,10 @@ private Row update(final Row original) {
         final DataContext dc = _updateCallback.getDataContext();
         final Table table = getTable();
         final List<FilterItem> whereItems = getWhereItems();
-        final DataSet dataSet = dc.query().from(table).select(table.getColumns()).where(whereItems).execute();
+        final DataSet dataSet = dc.query()
+                .from(table)
+                .select(table.getColumns().toArray(new Column[table.getColumns().size()]))
```

Comment:
This is a bit of a shame. No good way to avoid the extra `toArray()`? It seems you added an overriden method to `Query` that does the same, but I might be missing something here.